commit 93905dca4f4fe846e8098e24b3cb057e2b337128 (refs/changes/42/1542/5)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-02-27T17:32:22+00:00 (2 years, 7 months ago)
    
    TOOLS-1682 `sdcadm create -s SERVER ...` should support server hostname
    Reviewed by: Trent <trent.mick@joyent.com>

diff --git a/lib/cli/do_create.js b/lib/cli/do_create.js
index 4aa3f85..00e6a54 100644
--- a/lib/cli/do_create.js
+++ b/lib/cli/do_create.js
@@ -9,6 +9,7 @@
  */
 
 var p = console.log;
+var util = require('util');
 
 var assert = require('assert-plus');
 var vasync = require('vasync');
@@ -70,6 +71,25 @@ Create.prototype.execute = function cExecute(opts, args, cb) {
                 next(err);
             });
         },
+        function getServerFromHostname(_, next) {
+            if (!opts.server || common.UUID_RE.test(opts.server)) {
+                next();
+                return;
+            }
+            self.sdcadm.cnapi.listServers({
+                hostname: opts.server
+            }, function (err, servers) {
+                if (err || !servers.length) {
+                    self.sdcadm.log.error({err: err}, 'do_create');
+                    next(new errors.UsageError(util.format(
+                        'Cannot find server "%s"', opts.server)));
+                    return;
+                }
+                opts.server = servers[0].uuid;
+                next();
+            });
+
+        },
         function getChangeFromArgs(_, next) {
             if (args.length === 0) {
                 return next(new errors.UsageError(
@@ -93,7 +113,7 @@ Create.prototype.execute = function cExecute(opts, args, cb) {
 
             if (!opts.server) {
                 return next(new errors.UsageError(
-                    'Must specify server uuid'));
+                    'Must specify server uuid or hostname'));
             }
             change.server = opts.server;
             change.type = 'create-instance';
@@ -238,7 +258,7 @@ do_create.options = [
     {
         names: ['server', 's'],
         type: 'string',
-        help: 'The UUID for the target server.'
+        help: 'The server (UUID or hostname) on which to create the instance.'
     },
     {
         names: ['yes', 'y'],
